### ISSUE TITLE
Major bug with 4.7.12

### ISSUE THREAD

**Issue by Phooh** [Nagru/Manga-Organizer#8]
My entries preview images won't show up no matter what. Like MO just won't see the images as if the folder is empty. 

Another bug I found is, I decided to do a little maintenance, remove unused tags/artists, ect. after I decided to find missing sources some of my some of my entries are using a completely different path. Like for instances:
[Drive letter]:\folder\[Doujin name]
its turned into this:
[Drive letter]:\MangaDB.sqlite or MangaDatabase.bin_Deprecated(I can't remember which but its basicly trying use what's ever in the Manga Organizer folder as a replacement path.)
I can't seem to recreate this bug though, I'm guessing it had something to do with finding missing sources in the maintenance tab. I backed everything up so I went back to 4.7.1 and had no problems preforming maintenance.

**Comment by Nagru** [comment]
This is due to the similarity calculation when trying to find a missing file. If the source file specified doesn't exist the program will try to find one that's 70% similar under the root path. I've upped that to 80% and updated it to filter by file type so that it won't pick up on unrelated files.

**Comment by Nagru** [comment]
Ah, also I should mention that the auto path-correction for missing sources doesn't automatically save the new path. If a possible replacement path is found the save button will be enabled in case you want to save the change, or manually correct and then save.

**Comment by Phooh** [comment]
Cool, thanks for the updates. I wont have time to try 4.7.13 until later tonight though.

**Comment by Phooh** [comment]
I'm still seeing the same bugs. The entrie preview images still won't show up in the view tab. It seems to be random entries. Some show up, some don't. I checked my entries folders to see if it was a certain file type(.jpg, .png, ect.) to see if that was the problem but, I don't see any difference from the ones that show up and the ones that do.

As for the wong path bug, sorry I wasn't specific enough. Some entries would also give me and error saying that this entrie can't load and would give me the path to that entrie. Before(4.7.12) it would give me one of the MO database file as the error path but now it just gives me my normal path. I can still see the error entrie in the view tab but it just seems the same as the missing preview image bug. Also error entrie bug seem to be rarer than the missing preview image bug because once I click ok after the error shows up, I can reclick that same entrie and the error doesn't show up anymore.

tl;dr
I believe its mostly the missing preview image bug that's causing all these problems. It thinks the entrie folder is empty even when I try to re add the folder.

**Comment by Nagru** [comment]
I've <a href="https://cloud.openmailbox.org/public.php?service=files&t=54211c1ff8c748ba9dca4fe8b8398566&download" target="_blank">uploaded a version of the program</a> with all path correction removed. Would you run that and let me know if the behaviour is any different.

Also, would you let me know what your directory structure is (ex. "C:\Manga\\[Artist] Title\Pg01.jpg")? *And* which point in that path is being entered into MO. This really sounds like the program is struggling with the filepath and the auto-correction is either making it more confusing or exacerbating it.

**Comment by Nagru** [comment]
Wait, think I've got it. "<i>*.jp?g</i>" is the culprit; only affects Directory/Image and not Directory/Archive. The update will be pushed out this afternoon.

**Comment by Nagru** [comment]
All right, give the latest version a shot and let me know how it behaves.

**Comment by Phooh** [comment]
Just tested it, previews show up now. Thanks for your hard work on this. 

One thing also, when you mentioned the structure of my directory. I do have one bit of a different problem. I sort of shot myelf in the foot by having a really long directory path for all my manga/doujins. To the point where I someimes hit the character limit and have to rename my folders. For example:
F:\[Just imagine a bunch of different folders here]\Doujinshi\[Folders from A-Z]\artist names\[Then the manga/doujins inside]

I save all this on an external hard drive and was planning on one day moving all my manga and doujins to another hard drive. The problem is though I have over 5,000 entries in MO and I'd hate to have to move them all one by one to a new directory or re adding and tagging each one if I use the scan. I was wondering is possable to add or move multiple directories to their right folder path or something?

**Comment by Nagru** [comment]
Now that the DB is in SQLite there's a good way of handling this. After you copy your collection to the new HDD, open mangaDB.sqlite with <a href="http://sqlitebrowser.org/" target="_blank">this</a>. We can then update the recorded location of all your manga in one fell swoop. 

I don't know if you have any familiarity with SQL. If you don't just let me know what you expect the new path will be and I can write the update statement for you. (It should be easy to tweak later if you need to). This would be the preferred route anyways.

**Comment by Phooh** [comment]
I just downloaded sqlitebrowser. This is exactly what I needed. Been playing around with it and I'm slowly getting the gist of it. I can see where all my directories for each of my entries. I wouldn't mind a few examples on moving/changing multiple directories though. You can use these as an example:

C:\Doujinshi\A-C\[Artist]\[The doujin here]
C:\Doujinshi\D-F\[Artist]\[The doujin here]

You don't need to do each letter, that's just how my folder are.

**Comment by Nagru** [comment]
You'd want to use Replace() which has the syntax "replace(Value, 'Current', 'New')". All it does is swap out any text that matches 'Current' with the text 'New'. See the statement below for a practical example:
```
update Manga
set Location = replace(Location, 'F:\(the rest of your path)\Doujinshi\', 'C:\Doujinshi\')
```
Before you run that though, run this and check that the NewLocation results match what you expect:
```
select 
  Location
  ,replace(Location, 'F:\(the rest of your path)\Doujinshi\', 'C:\Doujinshi\') NewLocation
from Manga
```

**Comment by Phooh** [comment]
Everything works, I managed to get everything in working order(I understand SQL a bit better now). I'm still seeing that completely random rare false error message saying the entrie can't load but there's nothing wrong though(maybe something on my end). The entrie still loads fine and It's not causing any problems so I don't mind it.

I just wanna say thanks for the help and taking your time to fix my problems.

**Comment by Nagru** [comment]
No prob, thanks for letting me know there was a problem. 

I'd still like to fix that other issue though if you're up to it. It might be due to a cross-thread error. [This](https://cloud.openmailbox.org/public.php?service=files&t=54211c1ff8c748ba9dca4fe8b8398566) is a version of the program with a fix for that and more exception details.

Please give that a run and let me know what happens. If an error still pops up, screencap the message you see.

**Comment by Phooh** [comment]
Cool, I'll check it out later on.

**Comment by Phooh** [comment]
Haven't seen any errors for 2 days now.

**Comment by Nagru** [comment]
Sweet, sounds like we've got it. I've officially uploaded a new version then.

Thanks for the testing!
